infrastructure/src/main/java/org/corfudb/infrastructure/log/FileChannelPerf.java

```diff
@@ -0,0 +1,183 @@
+package org.corfudb.infrastructure.log;
+
+
+import com.codahale.metrics.Timer;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.ServerContext;
+import org.corfudb.util.MetricsUtils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class FileChannelPerf implements Closeable {//implements Closeable {
+
+    @Getter
+    static Timer writeMetrics;
+    @Getter
+    static Timer readMetrics;
+    @Getter
+    static Timer syncMetrics;
+
+    final static int MB_DATA = (1 << 20);
+    static final String READ_METRICS = "FileReadMetrics";
+    static final String WRITE_METRICS = "FileWriteMetrics";
+    static final String SYNC_METRICS = "FileSyncMetrics";
+    static boolean detectorEnabled;
+    static long spikeLatency;
+    static SlidingWindow slidingWindow;
+    private FileChannel fileChannel;
+
+
+    static public void setupParameters(int cntSpike, int maxLatencyVal, boolean report) {
+        detectorEnabled = report;
+        spikeLatency = maxLatencyVal;
+        readMetrics = ServerContext.getMetrics ().timer (READ_METRICS);
+        writeMetrics = ServerContext.getMetrics ().timer (WRITE_METRICS);
+        syncMetrics = ServerContext.getMetrics ().timer (SYNC_METRICS);
+        slidingWindow = new SlidingWindow(cntSpike, cntSpike, maxLatencyVal);
+    }
+
+    static private long getUnit(long size) {
+        return (size + MB_DATA - 1) / MB_DATA;
+    }
+
+    public FileChannelPerf(FileChannel fc) {
+        fileChannel = fc;
+    }
+
+    public static FileChannelPerf open(Path path, Set<? extends OpenOption> options,
+             FileAttribute<?>... attrs) throws IOException {
+        FileChannel fileChannel = FileChannel.open(path, options, attrs);
+        return new FileChannelPerf(fileChannel);
+    }
+
+    public static FileChannelPerf open(Path path, OpenOption... options) throws IOException {
+        FileChannel fileChannel = FileChannel.open(path, options);
+        return new FileChannelPerf(fileChannel);
+    }
+
+    public long size() throws IOException {
+        return fileChannel.size ();
```

Comment:
Can the space between the function and parenthesis be eliminated everywhere please?